soccer/gameplay/robocup-py.cpp

```diff
@@ -543,6 +546,52 @@ void Point_set_x(Geometry2d::Point* self, float x) { self->x() = x; }
 void Point_set_y(Geometry2d::Point* self, float y) { self->y() = y; }
 
 /**
+ * Given python function must take 2 float arguements, x and y, at the end of
+ * it's arguement
+ * declaration and must return a double / float
+ */
+float point_python_callback(Geometry2d::Point p, PyObject* pyfun) {
+    PyObject* pyresult =
+        PyObject_CallObject(pyfun, Py_BuildValue("ff", p.x(), p.y()));
+
+    if (pyresult == NULL) {
+        std::cerr << "Python callback function returned a bad value with args ";
+        std::cerr << p << std::endl;
+        return -1;
+    }
+
+    return PyFloat_AsDouble(pyresult);
+}
+
+boost::shared_ptr<std::function<float(Geometry2d::Point)>>
+stdfunction_constructor(PyObject* function) {
+    // Create aliased function to hid python function args
+    std::function<float(Geometry2d::Point)> f =
+        std::bind(&point_python_callback, std::placeholders::_1, function);
+
+    return boost::shared_ptr<std::function<float(Geometry2d::Point)>>(
+        new std::function<float(Geometry2d::Point)>(f));
+}
```

Comment:
Do I need to increment the reference counter for python functions if I'm going to be calling them in the "future"?